pkg/resource/unstructured/composed.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unstructured
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+)
+
+// ComposedOption modifies the composable resource.
+type ComposedOption func(resource *Composed)
+
+// FromReference sets the metadata of Composed.
+func FromReference(ref corev1.ObjectReference) ComposedOption {
+	return func(cr *Composed) {
+		cr.SetGroupVersionKind(ref.GroupVersionKind())
+		cr.SetName(ref.Name)
+		cr.SetNamespace(ref.Namespace)
+		cr.SetUID(ref.UID)
+	}
+}
+
+// NewComposed returns a new *Composed.
+func NewComposed(opts ...ComposedOption) *Composed {
+	cr := &Composed{}
+	for _, f := range opts {
+		f(cr)
+	}
+	return cr
+}
+
+// Composed is used to operate on the composable resources whose schema
+// is not known beforehand.
+type Composed struct {
+	unstructured.Unstructured
```

Comment:
Would it be worth embedding the `runtime.Unstructured` interface here instead?

https://godoc.org/k8s.io/apimachinery/pkg/runtime#Unstructured

Reply:
Ah nevermind - I forgot that interface doesn't satisfy `metav1.Object`. We could create a new interface that composes `metav1.Object` and `runtime.Unstructured`, but there doesn't seem to be much benefit in doing so.